lib/std/posix.zig

```diff
@@ -3402,6 +3402,24 @@ pub fn setregid(rgid: gid_t, egid: gid_t) SetIdError!void {
     }
 }
 
+pub const SetPgidError = error{
+    ProcessAlreadyExec,
+    InvalidProcessGroupID,
+    OperationNotPermitted,
+    NoSuchProcess,
+} || UnexpectedError;
+
+pub fn setpgid(pid: pid_t, pgid: pid_t) SetPgidError!void {
+    switch (errno(system.setpgid(pid, pgid))) {
+        .SUCCESS => return,
+        .ACCES => return error.ProcessAlreadyExec,
+        .INVAL => return error.InvalidProcessGroupID,
+        .PERM => return error.OperationNotPermitted,
+        .SRCH => return error.NoSuchProcess,
```

Comment:
Note that there is some precedence for the mapping/name of these errors that is not being followed here.

- In `setuid`/`seteuid`, `INVAL` is mapped to `error.InvalidUserId` (note the lowercase `d` at the end), so `error.InvalidProcessGroupID` should probably be `error.InvalidProcessGroupId` to match
- `PERM` is commonly mapped to `error.PermissionDenied`
- `SRCH` is commonly mapped to `error.ProcessNotFound`

No comment on the `ACCES` mapping, don't think there's any relevant precedence, so creating a new mapping seems like fair game.

(here's the description of the `EACCESS` error case for `setpgid` for context)

> The value of the pid argument matches the process ID of a child process of the calling process and the child process has successfully executed one of the [exec](https://pubs.opengroup.org/onlinepubs/009604599/functions/exec.html) functions.

Reply:
Thanks, I've renamed the errors accordingly.